### PR TITLE
CI: Modernise workflows

### PR DESCRIPTION
**Pull request by str4d**
We now check in `Cargo.lock` for testing against MSRV.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,54 +64,43 @@ jobs:
...
-      - uses: actions/checkout@v2
-      - name: Setup mdBook
-        uses: peaceiris/actions-mdbook@v1
-        with:
-          mdbook-version: '0.4.5'
+      - uses: actions/checkout@v4
+      - name: Install mdbook and mdbook-katex
+        run: |
+          mkdir mdbook
+          curl -sSL https://github.com/rust-lang/mdBook/releases/download/v0.4.35/mdbook-v0.4.35-x86_64-unknown-linux-gnu.tar.gz | tar -xz --directory=./mdbook
```
Comment (daira) [inline .github/workflows/ci.yml:71]:
```suggestion
          curl -sSL https://github.com/rust-lang/mdBook/releases/download/v0.4.35/mdbook-v0.4.35-x86_64-unknown-linux-gnu.tar.gz | tar -xz --directory=./mdbook
          curl -sSL https://github.com/lzanini/mdbook-katex/releases/download/v0.5.8pub/mdbook-katex-v0.5.8-x86_64-unknown-linux-gnu.tar.gz | tar -xz --directory=./mdbook
```

Comment (str4d) [reply .github/workflows/ci.yml:71]: We don't need the `mdbook-katex` plugin to run book tests, as the LaTeX segments are outside the `rust` code blocks.

Comment (str4d) [reply .github/workflows/ci.yml:71]: Oh wait, I see the actual bug: I copy-pasted the section from the book generator, and then removed the `mdbook-katex` line but forgot to change the name.

**Comment by str4d** [comment]
[Force-pushed](https://github.com/zcash/orchard/compare/09b4eb45d3c9344a869972a2ee503da827e531d2..a12661cf195e316ce7db8e02dd1267ad509e31df) to fix a copy-paste error.

**Review by daira — APPROVED** [review]
utACK

**Pull request merged**